builder/dockerfile/builder.go

```diff
@@ -151,19 +151,23 @@ func (b *Builder) build(source builder.Source, dockerfile *parser.Result) (*buil
 	addNodesForLabelOption(dockerfile.AST, b.options.Labels)
 
 	if err := checkDispatchDockerfile(dockerfile.AST); err != nil {
+		triggeredBuilds.WithValues("error_parsing_dockerfile").Inc()
```

Comment:
Looks like this should be decomposed into an "error" label and a "reason" label:

```
triggered_builds{"status"="error", "reason"="dockerfile_syntax"} 123
triggered_builds{"status"="error", "reason"="dockerfile_empty"} 123
triggered_builds{"status"="error", "reason"="fail"} 123 ; not sure about this one
triggered_builds{"status"="success"} 123
```